### PR TITLE
Fixes lot of things

### PR DESCRIPTION
**Pull request by tolbon**
Convert certain Tabs to Spaces,
Fix PHP Docs
Type declaration in function
Fix potential php errors

### CODE REVIEW

File: core/Period.php [hunk 1]
```diff
@@ -90,8 +90,8 @@ public function __wakeup()
      * etc.
      *
      * @static
-     * @param  $dateString string The **date** query parameter value.
-     * @param  $period string The **period** query parameter value.
+     * @param string $dateString string The **date** query parameter value.
+     * @param string $period string The **period** query parameter value.
```
Comment (sgiehl) [inline core/Period.php]:
```suggestion
     * @param string $dateString The **date** query parameter value.
     * @param string $period The **period** query parameter value.
```

File: libs/upgradephp/upgrade.php [hunk 2]
```diff
@@ -107,184 +112,194 @@
...
  *
- * @author Andrew Sohn <asohn (at) aircanopy (dot) net>
- * @author anthon (dot) pang (at) gmail (dot) com
- *
  * @param string $filename
  * @param bool $process_sections (defaults to false)
  * @return array
+ * @author anthon (dot) pang (at) gmail (dot) com
+ *
+ * @author Andrew Sohn <asohn (at) aircanopy (dot) net>
```
Comment (sgiehl) [inline libs/upgradephp/upgrade.php]: moving the author blocks and adding an empty line between them doesn't make things better...

File: libs/upgradephp/upgrade.php [hunk 3]
```diff
@@ -107,184 +112,194 @@
...
  *
- * @author BigueNique AT yahoo DOT ca
- * @author anthon (dot) pang (at) gmail (dot) com
- *
  * @param string $pattern
  * @param int $flags GLOBL_ONLYDIR, GLOB_MARK, GLOB_NOSORT (other flags not supported; defaults to 0)
  * @return array
+ * @author anthon (dot) pang (at) gmail (dot) com
+ *
+ * @author BigueNique AT yahoo DOT ca
```
Comment (sgiehl) [inline libs/upgradephp/upgrade.php]: same as above

File: plugins/CoreHome/DataTableRowAction/RowEvolution.php [hunk 4]
```diff
@@ -173,6 +173,8 @@ protected function loadEvolutionReport($column = false)
         }
 
         $isComparing = DataComparisonFilter::isCompareParamsPresent();
+        $unmodifiedSeriesLabels = [];
```
Comment (sgiehl) [inline plugins/CoreHome/DataTableRowAction/RowEvolution.php]: Guess the exact same line can then be removed a few lines below

File: plugins/CoreVisualizations/Visualizations/Sparklines.php [hunk 5]
```diff
@@ -126,13 +126,13 @@ private function fetchConfiguredSparklines()
 
         $originalDate = Common::getRequestVar('date');
         $originalPeriod = Common::getRequestVar('period');
-
+        $comparisonRows = [];
         if ($this->isComparing() && !empty($comparisons)) {
             $comparisonRows = [];
```
Comment (sgiehl) [inline plugins/CoreVisualizations/Visualizations/Sparklines.php]: That line can be removed then, I guess
```suggestion
```

File: plugins/Installation/FormFirstWebsiteSetup.php [hunk 6]
```diff
@@ -103,6 +103,7 @@ function validateOwner()
 
         // If intl extension is installed, get default currency from timezone country.
         if (!Option::get(API::OPTION_DEFAULT_CURRENCY) && $timezone && class_exists('NumberFormatter')) {
+            $location = [];
```
Comment (sgiehl) [inline plugins/Installation/FormFirstWebsiteSetup.php]: not necessarily needed, but doesn't hurt...

**Review by sgiehl — CHANGES_REQUESTED** [review]
Left a couple of comments. Not sure if any type hint might cause any problems, but we'll see if the tests are still passing...

**Comment by tolbon** [comment]
I see lot of failing compilation introduce by my modification I fix this asap

**Comment by tolbon** [comment]
I need help a test don't pass but I don't see where I need to revert 
https://travis-ci.org/matomo-org/matomo/jobs/657749446#L877
https://travis-ci.org/matomo-org/matomo/jobs/657749446#L925

I didn't find where this xml is generated

**Comment by sgiehl** [comment]
you can ignore the both tests mentioned. They are failing on 4.x-dev as well. Guess that's a sorting issue between PHP 7.2 and PHP 7.3 or something like that.

**Comment by tolbon** [comment]
I know it’s too late but ...
My commit is very big, Very hard for you to reviews this no ? Can I do something 
Do you want I close this PR and do more little  PR ?

**Comment by sgiehl** [comment]
Having smaller PRs makes it a lot easier to review and check why tests are failing. This PR has still a lot failing tests, but it's hard to say which change made them fail. So might be better to split the changes

**Comment by tolbon** [comment]
Yeah I close this and make little Pull Request, sorry for this

**Pull request closed without merging**